Tasks/UseNodeV1/installer.ts

```diff
@@ -38,36 +38,51 @@ interface INodeVersion {
 //      PATH = cacheDir + PATH
 //
 export async function getNode(versionSpec: string, checkLatest: boolean) {
+    let installedArch = osArch;
+
     if (toolLib.isExplicitVersion(versionSpec)) {
         checkLatest = false; // check latest doesn't make sense when explicit version
     }
 
     // check cache
     let toolPath: string;
     if (!checkLatest) {
-        toolPath = toolLib.findLocalTool('node', versionSpec, osArch);
+        toolPath = toolLib.findLocalTool('node', versionSpec, installedArch);
+
+        // In case if it's darwin arm and toolPath is empty trying to find x64 version
+        if (!toolPath && isDarwinArm(osPlat, installedArch)) {
+            toolPath = toolLib.findLocalTool('node', versionSpec, 'x64');
+        }
     }
 
     if (!toolPath) {
         let version: string;
         if (toolLib.isExplicitVersion(versionSpec)) {
             // version to download
             version = versionSpec;
-        }
-        else {
+        } else {
             // query nodejs.org for a matching version
-            version = await queryLatestMatch(versionSpec);
+            version = await queryLatestMatch(versionSpec, installedArch);
+
+            if (!version && isDarwinArm(osPlat, installedArch)) {
+                // nodejs.org does not have an arm64 build for macOS, so we fall back to x64
+                console.log(taskLib.loc('TryRosetta', osPlat, installedArch));
```

Comment:
It's better to use the debug, warning, etc. methods from task-lib to interact with stdout.

Reply:
`taskLib.debug(taskLib.loc('TryRosetta', osPlat, installedArch));`

Reply:
I'm using console log because this info message should display every time when we can't find arm version, so taskLib.debug is not working in this case